### PR TITLE
Add a forwardRef to the Checkbox component

### PR DESCRIPTION
**Pull request by jasperswart**
This PR:
- adds ref to Checkbox component
- includes unit test

### CODE REVIEW

File: packages/asc-ui/src/components/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -55,4 +58,6 @@ const Checkbox: React.FC<Props & React.HTMLAttributes<HTMLInputElement>> = ({
   )
 }
 
-export default Checkbox
+const CheckboxAny: any = Checkbox
```
Comment (jonkoops) [inline packages/asc-ui/src/components/Checkbox/Checkbox.tsx]: We do not have to use any here, we can directly wrap the component definition in a call to `forwardRef` in order to get all the type information we need:

```ts
const Checkbox = React.forwardRef<HTMLInputElement, Props & React.HTMLAttributes<HTMLInputElement>>(props => {

})

export default Checkbox
```

`forwardRef` takes two [generic types](https://www.typescriptlang.org/docs/handbook/generics.html), the first being the type of the element that is forwarded and the second the type of the props of the component. For more information check out this [blog post](https://medium.com/@martin_hotell/react-refs-with-typescript-a32d56c4d315).

File: packages/asc-ui/src/components/Checkbox/Checkbox.stories.tsx [hunk 2]
```diff
@@ -34,6 +36,9 @@ storiesOf('Atoms/Checkbox', module)
       <Label htmlFor="error" label="Error">
         <Checkbox id="error" error />
       </Label>
+      <Label htmlFor="ref" label="Ref">
```
Comment (jonkoops) [inline packages/asc-ui/src/components/Checkbox/Checkbox.stories.tsx]: Would it not be more appropriate to add a specific entry to the story? Perhaps more generic documentation can be placed in one of the markdown files.

Comment (jasperswart) [reply packages/asc-ui/src/components/Checkbox/Checkbox.stories.tsx]: yes I agree

File: packages/asc-ui/src/components/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (jasperswart) [reply packages/asc-ui/src/components/Checkbox/Checkbox.tsx]: gedaan

File: packages/asc-ui/src/components/Checkbox/Checkbox.stories.tsx (hunk 2 above)
Comment (jasperswart) [reply packages/asc-ui/src/components/Checkbox/Checkbox.stories.tsx]: which markdown file can I use then?

**Comment by jonkoops** [comment]
@jasperswart Let's keep this PR scoped to the checkbox component for now. We'll take care of the documentation later. I'd like to get this merged soon since I know some people are waiting for it.

File: packages/asc-ui/src/components/Checkbox/Checkbox.tsx [hunk 3]
```diff
@@ -7,15 +7,23 @@ import CheckboxStyle, {
...
-  className,
-  onChange,
-  variant,
-  disabled,
-  error,
-  ...otherProps
-}) => {
+type CheckboxProps = Props & React.HTMLAttributes<HTMLInputElement>
+
+const Checkbox: React.ForwardRefRenderFunction<
```
Comment (jonkoops) [inline packages/asc-ui/src/components/Checkbox/Checkbox.tsx]: We could just wrap the whole component here in a call to `forwardRef` so we can avoid having to pass the types twice.

```ts
const Checkbox = React.forwardRef<
  HTMLInputElement,
  Props & React.HTMLAttributes<HTMLInputElement>
>((
  {
    checked: checkedProp,
    className,
    onChange,
    variant,
    disabled,
    error,
    ...otherProps
  },
  ref,
) => {
  // ...
})

export default Checkbox
```

**Comment by jonkoops** [comment]
@jasperswart When you've implemented the changes could you squash your commits so we only have a single commit after merging?

File: packages/asc-ui/src/components/Checkbox/Checkbox.tsx (hunk 3 above)
Comment (chrisvanmook) [reply packages/asc-ui/src/components/Checkbox/Checkbox.tsx]: Agreed, this is more compact 👍

File: packages/asc-ui/src/components/Checkbox/Checkbox.stories.tsx (hunk 2 above)
Comment (jonkoops) [reply packages/asc-ui/src/components/Checkbox/Checkbox.stories.tsx]: Let's leave the documentation for now, I think we can do this in a separate PR.

File: packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx [hunk 4]
```diff
@@ -62,4 +62,16 @@ describe('Checkbox', () => {
...
+  it('should handle refs', () => {
+    const ref: any = React.createRef()
+
+    rerender(
+      <ThemeProvider theme={ascDefaultTheme}>
+        <Checkbox ref={ref} />
+      </ThemeProvider>,
+    )
+
+    expect(ref.current).toEqual(expect.any(Object))
```
[the remark this answers is not in the document]
Comment (jasperswart) [reply packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx]: thnks! daar was ik naar op zoek: gedaan

File: packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx [hunk 5]
```diff
@@ -62,4 +62,16 @@ describe('Checkbox', () => {
 
     expect(icon).not.toHaveStyleRule('background-color', expect.any(String))
   })
+
+  it('should handle refs', () => {
+    const ref: any = React.createRef()
```
[the remark this answers is not in the document]
Comment (jasperswart) [reply packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx]: gedaan

File: packages/asc-ui/src/components/Checkbox/Checkbox.stories.tsx [hunk 6]
```diff
@@ -3,6 +3,8 @@ import { storiesOf } from '@storybook/react'
 import Checkbox from './Checkbox'
 import Label from '../Label'
 
+const ref: React.RefObject<HTMLInputElement> = React.createRef()
```
[the remark this answers is not in the document]
Comment (jonkoops) [reply packages/asc-ui/src/components/Checkbox/Checkbox.stories.tsx]: Could you remove this for now?

File: packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx [hunk 7]
```diff
@@ -62,4 +62,17 @@ describe('Checkbox', () => {
 
     expect(icon).not.toHaveStyleRule('background-color', expect.any(String))
   })
+
+  it('should handle refs', () => {
+    const ref = React.createRef<HTMLInputElement>()
+
+    rerender(
```
Comment (jonkoops) [inline packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx]: I believe we can use `render` here instead of `rerender`

**Comment by jonkoops** [comment]
There are still a couple of comments on the PR that have not been resolved, @jasperswart could you take a look at them?

File: packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx (hunk 7 above)
Comment (jasperswart) [reply packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx]: that will result in code doubling

Comment (chrisvanmook) [reply packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx]: @jasperswart what do you mean?

File: packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx [hunk 8]
```diff
@@ -62,4 +63,19 @@ describe('Checkbox', () => {
 
     expect(icon).not.toHaveStyleRule('background-color', expect.any(String))
   })
+
+  it('should handle refs', () => {
+    const ref = React.createRef<HTMLInputElement>()
+
+    act(() => {
```
[the remark this answers is not in the document]
Comment (chrisvanmook) [reply packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx]: 🎉

File: packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx [hunk 9]
```diff
@@ -62,4 +63,19 @@ describe('Checkbox', () => {
...
+      rerender(
+        <ThemeProvider theme={ascDefaultTheme}>
+          <Checkbox ref={ref} />
+        </ThemeProvider>,
+      )
+    })
+
+    expect(ref.current).toBeInstanceOf(HTMLInputElement)
+    expect(ref.current && ref.current.type).toEqual('checkbox')
```
[the remark this answers is not in the document]
Comment (chrisvanmook) [reply packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx]: 🎉

File: packages/asc-ui/src/components/Checkbox/Checkbox.stories.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (jasperswart) [reply packages/asc-ui/src/components/Checkbox/Checkbox.stories.tsx]: what do you mean with that?

File: packages/asc-ui/src/components/Checkbox/Checkbox.tsx (hunk 3 above)
Comment (jasperswart) [reply packages/asc-ui/src/components/Checkbox/Checkbox.tsx]: gedaan

File: packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (jasperswart) [reply packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx]: gedaan

File: packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (jasperswart) [reply packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx]: gedaan

File: packages/asc-ui/src/components/Checkbox/Checkbox.stories.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (jonkoops) [reply packages/asc-ui/src/components/Checkbox/Checkbox.stories.tsx]: We're not documenting this for now, the ref will exist outside of the component which is not convenient for the person reading the documentation.

File: packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (jonkoops) [reply packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx]: The call to `act()` still needs to be removed.

[the remark this answers is not in the document]
Comment (jasperswart) [reply packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx]: gedaan

**Comment by jonkoops** [comment]
I've squashed the commits into a single one, the code looks good to me! Thanks @jasperswart

**Pull request merged**